### PR TITLE
feat(move to new demux flow)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR fully enables the new demultiplexing post processing flow in the `cg demultiplex finish flow-cell` and ``cg demultiplex finish all` commands.

The old demultiplexing post processing flow will still be accessible under `cg demultiplex finish flow-cell-old-flow` and `cg demultiplex finish all-old-flow

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by ChrOertlin** [comment]
# new commands are registered
image

# new cmds can be invoked
image

image


# finish all

only new flow run
image

# finish flow cell
only new flow
image

**Review by diitaz93 — APPROVED** [review]
:+1

**Comment by diitaz93** [comment]
Does this replace #2380 ?

**Review by islean — COMMENTED** [review]
Looks good! Although would we want to fix the "sequenced_at" so it is set to the flow cell's sequenced at if this is the new default command?

**Comment by ChrOertlin** [comment]
> Does this replace #2380 ?

No that one you mention will be on hold for now - but eventually that will be merged.

File: cg/cli/demultiplex/finish.py [hunk 1]
```diff
@@ -19,32 +19,25 @@ def finish_group():
     """Finish up after demultiplexing."""
 
 
-@finish_group.command(name="all")
+@finish_group.command(name="all-old-flow")
 @DRY_RUN
 @click.pass_obj
-def finish_all_cmd(context: CGConfig, dry_run: bool) -> None:
+def finish_all_cmd_old_flow(context: CGConfig, dry_run: bool) -> None:
     """Command to post-process all demultiplexed flow cells."""
```
Comment (henrikstranneheim) [inline cg/cli/demultiplex/finish.py:26]: Should w extend doc str to clarify what old means?

**Comment by ChrOertlin** [comment]
the flow cell sequenced at is set to date time now when it is parsed. The samples sequenced at is set to the datetime now soon thereafter. 

I think it is okay to leave it for now and improve after a refinement meeting on this.

**Review by henrikstranneheim — APPROVED** [review]
👍 . So nice to see this happening 🤩

File: cg/cli/demultiplex/finish.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/cli/demultiplex/finish.py]: I updated the docstring and mentions that it uses the old api.

**Review by karlnyr — APPROVED** [review]
neat enough

File: cg/cli/demultiplex/finish.py [hunk 2]
```diff
@@ -19,37 +19,32 @@ def finish_group():
...
     )
     demux_post_processing_api.set_dry_run(dry_run=dry_run)
     demux_post_processing_api.finish_all_flow_cells()
 
-    # Temporary finish flow cell logic will replace logic above when validated
-    demux_post_processing_api_temp: DemuxPostProcessingAPI = DemuxPostProcessingAPI(config=context)
-    demux_post_processing_api_temp.set_dry_run(dry_run=dry_run)
-    is_error_raised: bool = demux_post_processing_api_temp.finish_all_flow_cells_temp()
-    if is_error_raised:
-        raise click.Abort
```
Comment (Vince-janv) [inline cg/cli/demultiplex/finish.py:38]: In `demux_post_processing_api_temp.finish_all_flow_cells_temp()` we catch any error and return wether or not an error has occurred. Without the `raise click.Abort` in the end, we will not report any failures. I think a better solution can be found than the one we had, but we should exit with a failed exit code if an error has occured

Comment (ChrOertlin) [reply cg/cli/demultiplex/finish.py:38]: https://github.com/Clinical-Genomics/cg/pull/2389 addressed here!

**Comment by ChrOertlin** [comment]
stage:
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/demultiplexed-runs
done.
Log deploy... done.
cg, version 45.0.1

prod:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/demultiplexed-runs
done.
Log deploy... done.
cg, version 45.0.1

**Pull request merged**